atlasdb-commons/src/main/java/com/palantir/common/proxy/SingleCallProxy.java

```diff
@@ -1,59 +0,0 @@
-/*
- * (c) Copyright 2018 Palantir Technologies Inc. All rights reserved.
- *
- * Licensed under the Apache License, Version 2.0 (the "License");
- * you may not use this file except in compliance with the License.
- * You may obtain a copy of the License at
- *
- *     http://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
- */
-package com.palantir.common.proxy;
-
-import com.palantir.logsafe.exceptions.SafeIllegalStateException;
-import java.lang.reflect.InvocationTargetException;
-import java.lang.reflect.Method;
-import java.lang.reflect.Proxy;
-import java.util.concurrent.atomic.AtomicBoolean;
-
-/**
- * Ensure that only a single method is ever called.  Throw {@link IllegalStateException} on all subsequent calls.
- */
-public final class SingleCallProxy implements DelegatingInvocationHandler {
```

Comment:
Originally I planned to deprecate this, however it's not actually used outside of atlas.